front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/index.tsx

```diff
@@ -136,28 +136,70 @@ const InputsTable = ({
     setSelectedRows(new Set());
   }, [selectedCategory, pageNumber]);
 
+  // Reset input index when input no longer appears in table
   useEffect(() => {
-    if (!isNilOrError(inputs) && inputs.length === 0) {
-      setIsSideModalOpen(false);
+    if (!isNilOrError(inputs)) {
```

Comment:
I don't like the index reset here, it makes the not categorized section very annoying to use, I think we need to keep the index, but set in state that it doesn't match the previewed item, so when you click next you get to the input with the current index, and previous to the input for the preceding index.

Reply:
Hm, interesting thought! I do agree it's a bit annoying, will try your suggestion and see if I can make it happen.

Reply:
Went with adding the necessary logic to the component without relying on the cache - it was a bit tricky to avoid dependencies in the moveUp and moveDown functions but it works well now ⭐ 
Thanks again for brainstorming with me on this!